src/microbe_stage/ProcessSpeedInformation.cs

```diff
@@ -1,48 +1,42 @@
 ﻿using System.Collections.Generic;
+using System.Linq;
 
 /// <summary>
 ///   Speed information of a process in specific patch. Used in the
 ///   editor to show info to the player.
 /// </summary>
-public class ProcessSpeedInformation
+public class ProcessSpeedInformation : IProcessDisplayInfo
 {
+    public readonly Dictionary<Compound, float> WritableInputs = new Dictionary<Compound, float>();
+    public readonly Dictionary<Compound, float> WritableOutputs = new Dictionary<Compound, float>();
+    public readonly List<Compound> WritableLimitingCompounds = new List<Compound>();
+
+    public readonly Dictionary<Compound, float> AvailableAmounts = new Dictionary<Compound, float>();
+
+    // ReSharper disable once CollectionNeverQueried.Global
+    public readonly Dictionary<Compound, float> AvailableRates = new Dictionary<Compound, float>();
```

Comment:
these should be get only properties, right?

Reply:
No these are needed to be able being set in other places, will remove the readonlys.

Reply:
At least make them properties with public set, then.

Reply:
Actually, hang on. How did this code even work in the first place if these readonlies are set from other places? That can't compile.

Reply:
Wait, now that I think about it, making these get only properties makes sense as the actual assigning currently only happens here (hence the readonlys working). I must have confused the modifying of a dictionary keyvalue pairs with it requiring a setter, woops...